src/ContainerQueryCore.ts

```diff
@@ -7,21 +7,39 @@ export default class ContainerQueryCore {
   private rol: ResizeObserverLite;
   private result: Params = {};
 
-  constructor(query: Query, callback: (params: Params) => void) {
+  constructor(private query: Query, private callback: (params: Params) => void) {
     this.rol = new ResizeObserverLite((size) => {
-      const result = matchQueries(query)(size);
-      if (!isShallowEqual(this.result, result)) {
-        callback(result);
-        this.result = result;
-      }
+      this._processSize(size);
     });
   }
 
   observe(element: Element) {
-    this.rol.observe(element);
+    // Check its an Element node. It could be null, or a Text node if the props.children function returns a string
+    if (element instanceof Element) {
+      const styles = window.getComputedStyle(element);
+      this._processSize({
+        width: this._getNumber(styles.width),
+        height: this._getNumber(styles.height),
+      });
```

Comment:
Sorry for the late reply. 

Hmm, this is interesting. I don't fully understand. I have to investigate a bit. I thought once `observe` is called, the callback registered with `ResizeObserverLite` will be triggered immediately. But look like it's not.

Reply:
Yeah. It is interesting. I didn't look into `ResizeObserverLite` to see if there is a problem in there. I just assumed that if all the resize events were finished by the time the component was mounted then `ResizeObserverLite` would have no trigger to fire it.